### PR TITLE
GEM Timing Plots

### PR DESCRIPTION
**Pull request by nickh2000**
#### PR description:
This PR creates two GEM timing plots:

gemVFATBXPerChamber - plots hits by VFAT and BX, each plot indexed by different chambers and layers

gemChamberVFATBX - plots hits by VFAT and Chamber, each plot indexed by different BX's (-3 to 3)

#### PR validation:

Validated with ````runTheMatrix.py -l limited -i all --ibeos.````

### CODE REVIEW

**Comment by nickh2000** [comment]
https://github.com/cms-sw/cmssw/labels/code-checks-pending

**Comment by nickh2000** [comment]
https://github.com/cms-sw/cmssw/labels/code-checks-pending

**Comment by emanueleusai** [comment]
please test

**Comment by emanueleusai** [comment]
type gem

**Comment by emanueleusai** [comment]
type feature

**Comment by emanueleusai** [comment]
+1

**Comment by qliphy** [comment]
+1

File: DQM/L1TMonitor/src/L1TStage2EMTF.cc [hunk 1]
```diff
@@ -896,11 +955,22 @@ void L1TStage2EMTF::analyze(const edm::Event& e, const edm::EventSetup& c) {
...
+      int vfat = phi_part * 8 + Hit->Partition();
       if (Hit->Neighbor() == false) {
         gemChamberPad[hist_index]->Fill(chamber, Hit->Pad());
         gemChamberPartition[hist_index]->Fill(chamber, Hit->Partition());
         gemHitOccupancy->Fill(chamber_bin(1, 1, chamber), (endcap > 0) ? 1.5 : 0.5);  // follow CSC convention
-      } else {
+        //Added plots 07-21-22 ***
+        gemVFATBXPerChamber[chamber - 1][hist_index][layer]->Fill(Hit->BX(), vfat);
+        //indexed plots by BX 07-21-22
+        gemChamberVFATBX[hist_index][Hit->BX() + 3]->Fill(chamber_bin(1, 1, chamber), vfat);
```
Comment (qliphy) [inline DQM/L1TMonitor/src/L1TStage2EMTF.cc:969]: It seems "Hit->BX() + 3" can be as large as "7" and exceeds the boundary, which then cause the issue as reported in https://github.com/cms-sw/cmssw/issues/38940

**Review by qliphy — COMMENTED** [review]
@nickh2000 Please have a check and fix on the issue https://github.com/cms-sw/cmssw/issues/38940

**Comment by eyigitba** [comment]
Hi @qliphy , this is very strange indeed. EMTF DAQ readout window is [-3,+3] which should never return `7` for this. Maybe this protection should be added to the EMTF unpacker rather than the DQM plots.

**Comment by eyigitba** [comment]
ok, I found the problem. During LS2 there were some issues with corrupted GEM TPs which apparently sometimes extend to invalid BX values (4 in this case). I'll add a protection to the unpacker about this soon. This should fix this crash as well

**Pull request merged**